File: core/Device/Device.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import json

class Device():
    """Classe représentant le réveil"""

    scriptpath = "./component/Device/"
    config = json.load(open(scriptpath+'Device.json','r'))

    
    def __init__(self):
        """Constructeur de notre classe
        _connexion_mode =1 lan 2 wlan or 3 wlan_safe"""
        """data = open('device.json')
        x = json.load(data, object_hook=lambda d: Namespace(**d))"""

        self._name = Device.config['name']
        self._language = Device.config['language']
        self._connexionmode = Device.config['connexion_mode']
        self._timezone = Device.config['timezone']
        
    def _get_connexionmode(self):
        """méthode qui va lire l'attibut 'resultformat' depuis le fichier json du plugin"""
        return self._connexionmode
    
    def _get_timezone(self):
        """méthode qui va lire l'attibut 'timezone' depuis le fichier json du plugin"""
        return self._timezone

    def _set_connexionmode(self, connexionmode):
        """Méthode appelée quand on souhaite modifier le format du résultat"""

        Device.config["connexionmode"] = connexionmode
        with open(Device.scriptpath+'Device.json', 'w', encoding='utf-8') as f:
            json.dump(Device.config, f, indent=4)
        self._connexionmode = connexionmode


    connexionmode = property(_get_connexionmode,_set_connexionmode)
    timezone = property(_get_timezone)

File: core/Device/Device.php
<?php 

class Device{

    private $parsed_json_device ;
    
    
    private $name;
    private $language;
    private $theme_path;
    private $connexion_mode;    
    private $format_tt;    
    private $format_hour;    
    private $timezone;    

        
    public function get_name() {
        return $this -> name;
    }   
    public function get_language() {
        return $this -> $parsed_json_device->{'language'};
    }
    public function get_theme_path() {
        return $this -> theme_path;
    }   
    public function get_connexion_mode() {
        return $this -> connexion_mode;
    }
    public function get_format_tt() {
        return $this -> format_tt;
    }    
    public function get_format_hour() {
        return $this -> format_hour;
    }    
    public function get_timezone() {
        return $this -> timezone;
    }    
    public function set_name($name) {
        $this -> name = $name;
    }   
    public function set_language($language) {
        $this -> language =$language;
    }
    public function set_id_theme_path($theme_path) {
        $this -> theme_path = $theme_path;
    }
    public function set_connexion_mode($connexion_mode) {
        $this -> connexion_mode = $connexion_mode;
    }   
    public function set_format_tt($format_tt) {
        $this -> format_tt =$format_tt;
    }
    public function set_format_hour($format_hour) {
        $this -> format_hour = $format_hour;
    }
    public function set_timezone($timezone) {
        $this -> timezone = $timezone;
    }   

}


?>